helpers/testsuite/servers_test.go

```diff
@@ -0,0 +1,181 @@
+		if err != nil {
+			t.Log(err.Error())
+			t.FailNow()
+		}
+	}()
+	defer func() {
+		err = tcpServer.Kill()
+		if err != nil {
+			t.Log(err.Error())
+			t.FailNow()
+		}
+	}()
+
+	// === TEST: bundle using the test server ============================== //
+
+	newBundler, err := bundler.NewBundler(certFile, certFile)
+	checkError(err, t)
+	_, err = newBundler.BundleFromRemote("harryharpham.me", address, portTLS, bundler.Optimal)
+	checkError(err, t)
+
+	// === TEST: perform all of the default scans on the test server ======= //
+
+	results, err := scan.Default.RunScans(tlsServer.Addr, ".", ".")
+	checkError(err, t)
+	for family, familyResult := range results {
+		// We cannot test broad scans locally.
+		if family == "Broad" {
+			continue
+		}
+		for scanner, result := range familyResult {
+			if result.Error != "" {
+				t.Fatal("An error occurred during the following scan: " +
+					"[Family: " + family + ", Scanner: " + scanner +
+					", Grade: " + result.Grade + "]")
+			}
+			if result.Grade != scan.Good.String() {
+				t.Fatal("The following scan failed: [Family: " + family +
+					", Scanner: " + scanner + ", Grade: " + result.Grade + "]")
+			}
+		}
+	}
+
+	// === TEST: test the connectivity scan's response to a TCP-only server //
+
+	dnsLookup := scan.Connectivity.Scanners["DNSLookup"]
+	tcpDial := scan.Connectivity.Scanners["TCPDial"]
+	tlsDial := scan.Connectivity.Scanners["TLSDial"]
+
+	grade, _, err := dnsLookup.Scan(tcpServer.Addr)
+	if grade != scan.Good {
+		t.Log("TCP server failed DNSLookup scan (grade = " + grade.String() + ")")
+		t.FailNow()
+	}
+	checkError(err, t)
+
+	grade, _, err = tcpDial.Scan(tcpServer.Addr)
+	if grade != scan.Good {
+		t.Log("TCP server failed TCPDial scan (grade = " + grade.String() + ")")
+		t.FailNow()
+	}
+	checkError(err, t)
+
+	grade, _, err = tlsDial.Scan(tcpServer.Addr)
+	if grade != scan.Bad {
+		t.Log("TCP server did not fail TLSDial scan (grade = " + grade.String() +
+			", error = " + err.Error() + ")")
+		t.FailNow()
+	}
+}
+
+func TestStartCFSSLServer(t *testing.T) {
+	// We will test on this address and port. Be sure that these are free or
+	// the test will fail.
+
+	CACert, _, CAKey, err := initca.New(&CARequest)
+	checkError(err, t)
+
+	// Set up a test server using our CA certificate and key.
+	server := CFSSLServer{
+		Addr:  address,
+		Port:  portCFSSL,
+		CA:    CACert,
+		CAKey: CAKey,
+	}
+	err = server.Start()
+	checkError(err, t)
+
+	// Kill the server upon either completion or failure of the tests.
+	defer func() {
+		err = server.Kill()
+		checkError(err, t)
+	}()
+
+	// Try to start up a second server at the same address and port number. We
+	// should get an 'address in use' error.
+	server2 := CFSSLServer{
+		Addr:  server.Addr,
+		Port:  server.Port,
+		CA:    CACert,
+		CAKey: CAKey,
+	}
+	err = server2.Start()
+	if err == nil || !strings.Contains(err.Error(), "Error occurred on server: address") {
+		t.Fatal("Two servers allowed on same address and port.")
```

Comment:
I think it is ok not to test this. This is causing race condition in the tests and we probably never start two servers in the functional tests.